src/runtime/propensity-server.js

```diff
@@ -0,0 +1,262 @@
+      400: 'Invalid request',
+      500: 'Server not available',
+    };
+    /** @private @const {!XhrInterface} */
+    this.xhrInterface_ = new XhrInterface(errorMap);
+  }
+
+  /**
+   * @return {string}
+   */
+  getUrl_() {
+    if (this.TEST_SERVERS_) {
+      return 'http://sohanirao.mtv.corp.google.com:8080';
+    } else {
+      return 'https://pubads.g.doubleclick.net/gampad/adx';
+    }
+  }
+
+  /**
+   * @return {?string}
+   */
+  getGads_() {
+    let gads = null;
+    const gadsmatch = this.win_.document.cookie.match(
+        '(^|;)\\s*__gads\\s*=\\s*([^;]+)');
+    gads = gadsmatch && encodeURIComponent(gadsmatch.pop());
+    return gads;
+  }
+
+  /**
+   * Returns the client ID to be used
+   * @return {?string}
+   * @private
+   */
+  getClientId_() {
+    if (!this.clientId_) {
+      if (!this.userConsent_) {
+        this.clientId_ = 'noConsent';
+      } else {
+        this.clientId_ = this.getGads_();
+      }
+    }
+    return this.clientId_;
+  }
+
+  /**
+   * @param {boolean} userConsent
+   */
+  setUserConsent(userConsent) {
+    this.userConsent_ = userConsent;
+  }
+
+  /**
+   * @param {string} state
+   * @param {?string} entitlements
+   */
+  sendSubscriptionState(state, entitlements) {
+    this.product_ = entitlements;
+    const init = /** @type {!../utils/xhr.FetchInitDef} */ ({
+      method: 'GET',
+      credentials: 'include',
+      headers: {'Accept': 'text/plain, application/json'},
+    });
+    const clientId = this.getClientId_();
+    let url = this.getUrl_() + '/subopt/data?states=' + this.publicationId_
+        + ':' + state;
+    if (this.product_) {
+      url = url + ':' + this.product_;
+    }
+    if (clientId) {
+      url = url + '&cookie=' + clientId;
+    }
+    url = url + '&u_tz=240';
+    return this.xhrInterface_.sendRequest(url, init);
+  }
+
+  /**
+   * @param {string} event
+   * @param {?string} context
+   */
+  sendEvent(event, context) {
+    const init = /** @type {!../utils/xhr.FetchInitDef} */ ({
+      method: 'GET',
+      credentials: 'include',
+      headers: {'Accept': 'text/plain, application/json'},
+    });
+    const clientId = this.getClientId_();
+    let url = this.getUrl_() + '/subopt/data?events=' + this.publicationId_
+        + ':' + event;
+    if (context) {
+      url = url + ':' + JSON.stringify(context);
+    }
+    if (clientId) {
+      url = url + '&cookie=' + clientId;
+    }
+    url = url + '&u_tz=240';
+    return this.xhrInterface_.sendRequest(url, init);
+  }
+
+  /**
+   * @param {string} referrer
+   * @param {string} type
+   * @return {?Promise<JsonObject>}
+   */
+  getPropensity(referrer, type) {
+    const init = /** @type {!../utils/xhr.FetchInitDef} */ ({
+      method: 'GET',
+      credentials: 'include',
+      headers: {'Accept': 'text/plain, application/json'},
+    });
+    const clientId = this.getClientId_();
+    let url = this.getUrl_() + '/subopt/pts?products=' + this.publicationId_
```

Comment:
let url = this.getUrl_() + '/subopt/pts?products=' + this.publicationId_
        + '&type=' + type + '&u_tz=240'
        + '&ref=' + referrer;

Reply:
done